sqlglot/optimizer/unnest_subqueries.py

```diff
@@ -105,7 +105,7 @@ def unnest(select, parent_select, next_alias_name):
                 .from_(select.subquery("_q", copy=False), copy=False)
                 .group_by(exp.column(value.alias, "_q"), copy=False)
             )
-    else:
+    elif not isinstance(value.this, exp.AggFunc):
```

Comment:
We need to generalize this as follows:

```suggestion
    elif not value.this.find(exp.AggFunc):
```

Otherwise we'll still optimize queries like the following incorrectly (`value.this` will be an `Add` but there's still going to be an aggregate under that node):

```sql
SELECT s.t FROM s WHERE 1 IN (SELECT MAX(t.a) as t1 FROM t)
```

Reply:
+1 to @georgesittas 's comments, but use `find_in_scope` instead of `Expression.find`